### ISSUE TITLE
Support for `add_pooling_layer` in transformers.TFBertModel

### ISSUE THREAD

**Issue by tomigee** [huggingface/transformers#29567]
### Feature request

HuggingFace transformers currently support the `add_pooling_layer` argument in the initialization of a `transformers.BertModel` object [as shown here](https://huggingface.co/docs/transformers/model_doc/bert#transformers.BertModel), supported by PyTorch. My request is simply to extend this support to `transformers.BertModel`'s Tensorflow counterpart, `transformers.TFBertModel`.

### Motivation

I'm working on a model right now that leverages the pretrained `bert-base-uncased` model as an encoder to generate features, which are in turn fed into a decoder via cross-attention. As it currently stands, the TF implementation of `bert-base-uncased` available via HuggingFace has a pooling layer that condenses its output from shape `(batch, seq_length, hidden_dim)` to `(batch, 1, hidden_dim)`. While this is fine for simpler models that only attach an output head, it's problematic for more complex models in at least two ways:

1. The dimension reduction very likely leads to information loss within the model
2. When connecting this model to a decoder via cross-attention, it would be nice if the decoder could attend to individual tokens as it sees fit, as opposed to one, singular,  pooled token. In fact this was the implementation of the very first transformer from "Attention is All You Need", and is a rather important detail in the implementation in my opinion.

Given that `transformers.BertModel` already supports this, I'm hoping this is a rather easy implementation.

### Your contribution

Yes, I'm happy to help. Just let me know if this is a feature that you folks are willing to include, and if you'd like my help in doing so.

**Comment by amyeroberts** [comment]
Hi @tomigee, 

Sounds like a good idea - we want our TF and pytorch models to be as similar as possible. Feel free to open a PR and tag me and @Rocketknight1 for review!

**Comment by Rocketknight1** [comment]
Yes, agreed! This is an oversight, and it's definitely worth fixing. Note that the `TFBertMainLayer` class already supports this argument, so all you need to do is ensure that it gets correctly passed from `TFBertModel`.

**Comment by tomigee** [comment]
Done, see #29603
